disco/preprocess/select_timepoints2.py

```diff
@@ -166,166 +310,247 @@ def get_profile_data():
         profile_files: a list of paths pointing to shape (profile) files such as "LoadShapes.dss"
     OUTPUT:
-        bus_data: 
+        bus_data:
         profile_data:
         ag_series:
-        head_time_indices: list of critical time indices when only feeder-head timeseries are considered 
+        head_time_indices: list of critical time indices when only feeder-head timeseries are considered
         critical_time_indices: all critical time indices (individual buses as well as feeder-head considered)
         compression_rate: ratio between the number of critical timepoints and the total number of timepoints in the timeseries
-                             
+
     """
-    folder, _ = os.path.split(path_to_master)
-    if destination_dir==None:
-        folder, _ = os.path.split(path_to_master)
-        destination_dir = folder
-    else:
-        os.makedirs(destination_dir, exist_ok=True)
-    
-    bus_data={}
-    load_feeder(path_to_master)
-    for category, param_classes in category_class_dict.items():
+    if not destination_dir.exists():
+        raise FileNotFoundError(f"destination_dir={destination_dir} does not exist")
+
+    bus_data = {}
+    load_feeder(path_to_master, destination_dir, fix_master_file)
+    for category, param_classes in categories.items():
         for param_class in param_classes:
             bus_data = get_param_values(param_class, bus_data, category)
     profile_data = get_profile_data()
-    ag_series, head_time_indices, critical_time_indices, compression_rate = \
-        agregate_series(bus_data, 
-                        profile_data, 
-                        critical_conditions, 
-                        recreate_profiles, 
-                        destination_dir)
-    return bus_data, profile_data, ag_series, head_time_indices, critical_time_indices, compression_rate
+    ag_series, head_time_indices, critical_time_indices, compression_rate = aggregate_series(
+        bus_data,
+        profile_data,
+        critical_conditions,
+        recreate_profiles,
+        destination_dir,
+        create_new_circuit,
+    )
     
+    metadata_df = get_metadata(ag_series, head_time_indices)
+    metadata_df.to_csv(destination_dir / "metadata.csv")
+
+    logger.info("head_time_indices = %s length = %s", head_time_indices, len(head_time_indices))
+    logger.info(
+        "critical_time_indices = %s length = %s", critical_time_indices, len(critical_time_indices)
+    )
+
+    return (
+        bus_data,
+        profile_data,
+        ag_series,
+        head_time_indices,
+        critical_time_indices,
+        compression_rate,
+    )
+
 
 if __name__ == "__main__":
-    path_to_master = r"C:\Users\KSEDZRO\Documents\Projects\LA-equity-resilience\data\P12U\sb9_p12uhs3_1247_trans_264--p12udt8475\Master.dss"
-    destination = r"C:\Users\KSEDZRO\Documents\Projects\LA-equity-resilience\data\P12U"
+    path_to_master = Path(
```

Comment:
should this be removed from this file?

Reply:
I defer to @KwamiS on whether he wants to make users change this if they want to use it. This could be changed to use an OpenDSS feeder in the disco repo and avoid his hard-coded filesystem paths.